src/Apps/Order2/Routes/Checkout/Components/Order2CheckoutPricingBreakdown.tsx

```diff
@@ -46,7 +62,13 @@ export const Order2CheckoutPricingBreakdown: React.FC<
               amountText = "Waiting for offer"
             }
 
-            if (isValidTimer(partnerOffer)) {
+            // Show timer if we have a valid partner offer with time remaining
+            if (
+              partnerOffer &&
+              !timer.isExpired &&
+              !timer.isLoading &&
+              !timer.remainingTime.startsWith("NaN")
```

Comment:
Are all these checks neessessary? Maybe we could we refactor the hook so that it handles the invalid or missing times instead?